### PR TITLE
Add documentation for smithy-trait-package plugin

### PR DESCRIPTION
**Pull request by hpmellema**
#### Background
Adds documentation for the `smithy-trait-package` plugin including: 
1. Readme entries
2. Update examples descriptions
3. Updates to `smithy-examples.json` to make examples available via `smithy init` command

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -78,6 +81,33 @@ plugins {
...
+The `smithy-trait-package` plugin can be applied on its own to easily configure a trait package. 
+
+```kotlin 
+// build.gradle.kts
+plugins {
+    id("software.amazon.smithy.gradle.smithy-trait-package").version("1.1.0")
+}
+```
+
+*Note*: To use the `trait-codegen` Smithy build plugin to generate java trait definitions with this gradle plugin,
```
Comment (kstich) [inline README.md:103]:
```suggestion
*Note*: To use the `trait-codegen` Smithy build plugin to generate Java trait definitions with this Gradle plugin,
```

File: smithy-templates.json [hunk 2]
```diff
@@ -130,6 +130,26 @@
...
+      "documentation": "Trait package for simple Smithy structure trait with multiple inputs.",
+      "path": "examples/trait-package-plugin/create-simple-trait",
+      "include": [
+        "gradle/",
+        "gradlew",
+        "gradlew.bat",
+        ".gitignore"
+      ]
+    },
+    "existing-trait": {
```
Comment (kstich) [inline smithy-templates.json:144]: Q: do we have a generated trait with custom validation example?

File: README.md [hunk 3]
```diff
@@ -9,6 +9,9 @@ and generate JARs that contain filtered projections of Smithy models.
   the base `smithyBuild` task for the project that builds the Smithy models in the project.
 - [`smithy-jar` plugin](#smithy-jar-plugin): Adds built Smithy files to an existing `jar` task such as that created by the Java or Kotlin plugins. 
   The `smithy-jar` plugin also adds build metadata and tags to the JAR's MANIFEST. The `smithy-jar` plugin applies the `smithy-base` plugin when it is applied.
+- [`smithy-trait-package`](#smithy-trait-package-plugin): Configures a Java project for a custom trait definition. This plugin can be used in conjunction 
+  with the [`trait-codegen`](https://github.com/smithy-lang/smithy/tree/main/smithy-trait-codegen) smithy build plugin to generate Java accessors from Smithy trait definitions.
```
Comment (kstich) [inline README.md:13]: "accessors" seems like the wrong word, maybe "a Java representation of a Smithy trait definition"?

**Review by kstich — CHANGES_REQUESTED** [review]

File: smithy-templates.json (hunk 2 above)
Comment (hpmellema) [reply smithy-templates.json:144]: We do. In the smithy examples package.

**Review by kstich — APPROVED** [review]

**Pull request merged**